program/src/initialize_ncn_reward_router.rs

```diff
@@ -0,0 +1,108 @@
+use std::mem::size_of;
+
+use jito_bytemuck::{AccountDeserialize, Discriminator};
+use jito_jsm_core::{
+    create_account,
+    loader::{load_signer, load_system_account, load_system_program},
+};
+use jito_restaking_core::{config::Config, ncn::Ncn, operator::Operator};
+use jito_tip_router_core::{
+    ballot_box::BallotBox, loaders::load_ncn_epoch, ncn_fee_group::NcnFeeGroup,
+    ncn_reward_router::NcnRewardRouter,
+};
+use solana_program::{
+    account_info::AccountInfo, clock::Clock, entrypoint::ProgramResult, msg,
+    program_error::ProgramError, pubkey::Pubkey, rent::Rent, sysvar::Sysvar,
+};
+
+/// Can be backfilled for previous epochs
+pub fn process_initialize_ncn_reward_router(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    ncn_fee_group: u8,
+    first_slot_of_ncn_epoch: Option<u64>,
+) -> ProgramResult {
+    let [restaking_config, ncn, operator, ballot_box, ncn_reward_router, payer, restaking_program, system_program] =
+        accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    if restaking_program.key.ne(&jito_restaking_program::id()) {
+        msg!("Incorrect restaking program ID");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    Config::load(restaking_program.key, restaking_config, false)?;
+    Ncn::load(restaking_program.key, ncn, false)?;
+    Operator::load(restaking_program.key, operator, false)?;
+
+    load_system_account(ncn_reward_router, true)?;
+    load_system_program(system_program)?;
+    load_signer(payer, true)?;
+
+    let ncn_fee_group = NcnFeeGroup::try_from(ncn_fee_group)?;
+
+    let current_slot = Clock::get()?.slot;
+    let (ncn_epoch, _) = load_ncn_epoch(restaking_config, current_slot, first_slot_of_ncn_epoch)?;
+
+    let has_winning_ballot = {
+        let ballot_box_data = ballot_box.data.borrow();
+        let ballot_box = BallotBox::try_from_slice_unchecked(&ballot_box_data)?;
+        ballot_box.has_winning_ballot()
+    };
+
+    if !has_winning_ballot {
+        msg!("Ballot has to be finalized before initializing ncn reward router");
+        return Err(ProgramError::InvalidAccountData);
+    }
```

Comment:
could we get rid of this check?

Reply:
We can, however, closing this account will need the ballot box

Reply:
taking out for now